mos_tests/functions/common.py

```diff
@@ -409,3 +420,37 @@ def check_volume_status(cinderclient, uid, status, timeout=5):
         if inst_status == status:
             return True
     return False
+
+
+def check_volume_snapshot_status(cinderclient, uid, status, timeout=5):
+    """ Check status of volume
+            :param cinderclient: Cinder API client connection point
+            :param uid: UID of volume snapshot
+            :param status: Expected volume snapshot status
+            :param timeout: Timeout for check operation
+            :return True or False
+    """
+    if check_volume_snapshot(cinderclient, uid):
+        start_time = time()
+        snapshot_status = [s.status
+                           for s in cinderclient.volume_snapshots.list()
+                           if s.id == uid.id][0]
+        while snapshot_status != status and time() < start_time + 60 * timeout:
```

Comment:
while time() < start_time + 60 * timeout:
    sleep(1)
    snapshot_status = [s.status
                                   for s in cinderclient.volume_snapshots.list()
                                   if s.id == uid.id][0]
    if snapshot_status == status:
        return True